### PR TITLE
Backport #28817 to 21.8: Fix rare replicas diverge

### PR DESCRIPTION
**Pull request by robot-clickhouse**
Original pull-request #28817
Cherry-pick pull-request #29142

This pull-request is a last step of an automated backporting.
Treat it as a standard pull-request: look at the checks and resolve conflicts.
Merge it only if you intend to backport changes to the target branch, otherwise just close it.

### CODE REVIEW

File: src/Storages/MergeTree/ReplicatedMergeTreeRestartingThread.cpp [hunk 1]
```diff
@@ -168,12 +168,14 @@ bool ReplicatedMergeTreeRestartingThread::tryStartup()
...
 
+        storage.queue.initialize(zookeeper);
+
         storage.queue.load(zookeeper);
 
         /// pullLogsToQueue() after we mark replica 'is_active' (and after we repair if it was lost);
         /// because cleanup_thread doesn't delete log_pointer of active replicas.
         storage.queue.pullLogsToQueue(zookeeper, {}, ReplicatedMergeTreeQueue::LOAD);
-        storage.queue.removeCurrentPartsFromMutations();
+        storage.queue.removeCurrentPartsFromMutations
```
Comment (xjewer) [inline src/Storages/MergeTree/ReplicatedMergeTreeRestartingThread.cpp:178]: 🚓🚨

**Comment by alexey-milovidov** [comment]
Normally we should not backport anything that is non-trivial.

**Comment by alesapin** [comment]
@Mergifyio update

**Comment by alesapin** [comment]
@mergifyio update

**Comment by vitlibar** [comment]
@mergifyio update

**Comment by alexey-milovidov** [comment]
> Fix extremely rare case ... all replicas reboot

**Pull request closed without merging**